### PR TITLE
Add KakDecomposition._decompose_ by fixing factor of 2 error in exp(PauliString)

### PR DESCRIPTION
**Pull request by Strilanc**
- I compared against results from WolframAlpha to make sure the modified tests are now correct

### CODE REVIEW

File: cirq/ops/pauli_string_test.py [hunk 1]
```diff
@@ -532,6 +533,21 @@ def test_pow():
...
+def test_rpow():
+    a, b = cirq.LineQubit.range(2)
+
+    u = cirq.unitary(np.exp(1j * np.pi / 2 * cirq.Z(a) * cirq.Z(b)))
+    np.testing.assert_allclose(u, np.diag([1j, -1j, -1j, 1j]), atol=1e-8)
+
+    u = cirq.unitary(np.exp(-1j * np.pi / 4 * cirq.Z(a) * cirq.Z(b)))
+    cirq.testing.assert_allclose_up_to_global_phase(u,
+                                                    np.diag([1, 1j, 1j, 1]),
+                                                    atol=1e-8)
```
Comment (viathor) [inline cirq/ops/pauli_string_test.py:545]: Stronger assertion:

```python
cirq.testing. assert_allclose(u, np.diag(np.sqrt(-1j), np.sqrt(1j), np.sqrt(1j), np.sqrt(-1j)), atol=1e-8)
```

File: cirq/linalg/decompositions_test.py [hunk 2]
```diff
@@ -724,3 +724,16 @@ def test_kak_vector_negative_atol():
 def test_kak_vector_input_not_unitary():
     with pytest.raises(ValueError, match='must correspond to'):
         cirq.kak_vector(np.zeros((4, 4)))
+
+
+@pytest.mark.parametrize('unitary', [
+    cirq.testing.random_unitary(4),
```
Comment (viathor) [inline cirq/linalg/decompositions_test.py:730]: Optional: I wonder what distribution we get here on the elements of KAK (e.g. on the three interaction coefficients). A good addition to the random unitary test would be one synthesized from known coefficients, e.g.

```python
u = np.exp(1j*np.pi*(0.2*cirq.unitary(cirq.XX) + 0.3*cirq.unitary(cirq.YY) + 0.4*cirq.unitary(cirq.ZZ)))
```

File: cirq/ops/pauli_string_test.py (hunk 1 above)
Comment (Strilanc) [reply cirq/ops/pauli_string_test.py:545]: I intentionally only did it up to global phase to make the test simpler to write, to make sure the complications weren't hiding a mistake.

File: cirq/linalg/decompositions_test.py (hunk 2 above)
Comment (Strilanc) [reply cirq/linalg/decompositions_test.py:730]: I added a couple more cases. The "arbitrary fixed coefficients" case is very well covered by the random unitary; the corner cases have to do with interaction coefficients being zero or maxed.

**Pull request merged**